YSI_Visual/y_interiors/y_interiors_impl.inc

```diff
@@ -0,0 +1,300 @@
+	gYSI_InteriorData[gYSI_Interiorid][e_YSI_INTEXIT_Z] = YSI_zExit;
+	gYSI_InteriorData[gYSI_Interiorid][e_YSI_INTERIOR_INTID] = YSI_eInteriorid;
+	gYSI_InteriorData[gYSI_Interiorid][e_YSI_INTERIOR_INTID_OUT] = YSI_xInteriorid;
+	gYSI_InteriorData[gYSI_Interiorid][e_YSI_INTERIOR_VWID] = YSI_eVirtualWorldid;
+	gYSI_InteriorData[gYSI_Interiorid][e_YSI_INTERIOR_VWID_OUT] = YSI_xVirtualWorldid;
+
+	new YSI_EntranceLabelString[300];
+	format(YSI_EntranceLabelString,sizeof(YSI_EntranceLabelString),"{ffffff}[ %s ]\n"col_white"Enter here!",YSI_InteriorName);
+
+	Interiors_Create3DLabelAndPickup(YSI_EntranceLabelString, YSI_xEntrance,YSI_yEntrance,YSI_zEntrance, -1, -1, 19133, 2.0);
+
+	Interiors_Create3DLabelAndPickup("Exit here!", YSI_xExit,YSI_yExit,YSI_zExit, -1, -1, 19133, 2.0);
+	return YSI_INTERIORS_TRUE;
+}
+
+stock bool:Interiors_IsPlayerInInterior(playerid)
+{
+	if(gYSI_PlayersInterior[ playerid ] = -1) return false;
+	else return true;
+}
+
+stock Interiors_ClearPlayerInteriorData(playerid)
+{
+	gYSI_PlayersInterior[playerid] = -1;
+	return YSI_INTERIORS_TRUE;
+}
+
+stock Interiors_SetInteriorAngles(
+		const YSI_CustomInteriorid,
+		Float:YSI_aAngle,
+		Float:YSI_bAngle)
+{
+	gYSI_InteriorData[YSI_CustomInteriorid][e_YSI_INTENTRANCE_A] = YSI_aAngle;
+	gYSI_InteriorData[YSI_CustomInteriorid][e_YSI_INTEXIT_A] = YSI_bAngle;
+	return YSI_INTERIORS_TRUE;
+}
+
+/*
+
+                                                                                                 
+88888888888                                                          db         88888888ba   88  
+88                          ,d                                      d88b        88      "8b  88  
+88                          88                                     d8'`8b       88      ,8P  88  
+88aaaaa      8b,     ,d8  MM88MMM  8b,dPPYba,  ,adPPYYba,         d8'  `8b      88aaaaaa8P'  88  
+88"""""       `Y8, ,8P'     88     88P'   "Y8  ""     `Y8        d8YaaaaY8b     88""""""'    88  
+88              )888(       88     88          ,adPPPPP88       d8""""""""8b    88           88  
+88            ,d8" "8b,     88,    88          88,    ,88      d8'        `8b   88           88  
+88888888888  8P'     `Y8    "Y888  88          `"8bbdP"Y8     d8'          `8b  88           88  
+                                                                                                 
+                                                                                                 
+*/
+
+#define interiorcall%0(%1) \
+	forward%0@_yI(%1,YSI_Actionid); \
+	public%0@_yI(%1,YSI_Actionid)
+
+#define hookinteriorcall%0(%1) \
+	hook%0@_yI(%1,YSI_Actionid)
+
+stock Interiors_SetPlayerInterior(YSI_p_Playerid, pYSI_Interiorid)
+{
+	if(pYSI_Interiorid < -1)
+		return YSI_INTERIORS_FALSE;
+
+	gYSI_PlayersInterior[YSI_p_Playerid] = pYSI_Interiorid;
+	return YSI_INTERIORS_TRUE;
+}
+
+#define Interiors_InteriorActionPerformed(%0) \
+	(YSI_Actionid == %0)

```

Comment:
This is very poor practice.  This call will only work in some functions, for no apparent reason.  Don't hide variables like this.